src/System.Private.Xml/src/System/Xml/XmlCharType.cs

```diff
@@ -2,22 +2,73 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+//#define XMLCHARTYPE_USE_RESOURCE    // load the character properties from resources (XmlCharType.bin must be linked to System.Xml.dll)
+//#define XMLCHARTYPE_GEN_RESOURCE    // generate the character properties into XmlCharType.bin
+
+#if XMLCHARTYPE_GEN_RESOURCE || XMLCHARTYPE_USE_RESOURCE
+using System.IO;
+using System.Reflection;
+#endif
 using System.Threading;
 using System.Diagnostics;
-using System.Runtime.InteropServices;
-using System.Runtime.CompilerServices;
 
 namespace System.Xml
 {
-    internal unsafe struct XmlCharType
+
+    /// <include file='doc\XmlCharType.uex' path='docs/doc[@for="XmlCharType"]/*' />
+    /// <internalonly/>
+    /// <devdoc>
+    ///  The XmlCharType class is used for quick character type recognition
+    ///  which is optimized for the first 127 ascii characters.
+    /// </devdoc>
+#if SILVERLIGHT
+#if !SILVERLIGHT_XPATH
+    [System.Runtime.CompilerServices.FriendAccessAllowed] // Used by System.ServiceModel.dll and System.Runtime.Serialization.dll
+#endif
+#endif
+#if XMLCHARTYPE_USE_RESOURCE
+    unsafe internal struct XmlCharType {
+#else
+    internal struct XmlCharType
```

Comment:
You may want to run CodeFormatter.exe against this file, which is not following the corefx coding style.